### PR TITLE
formatter: implements export-as functionality

### PR DESCRIPTION
**Pull request by ioannistsanaktsidis**
* Implements select all results functionality.

Signed-off-by: Ioannis Tsanaktsidis <ioannis.tsanaktsidis@cern.ch>

### CODE REVIEW

File: inspire/base/static/js/search/search_results.js [hunk 1]
```diff
@@ -44,8 +44,215 @@ require(['jquery', 'bootstrap'], function($) {
...
+    sList = [];
+    $('#alert-selection').remove();
+    $("#download").removeAttr('href');
+    $('.checkbox-results').each(function() {
+      this.checked = false;
+    });
+    $('#checkbox-select-all').prop('checked', false);
+  });
+
+  function querystring(key) {
```
Comment (jmartinm) [inline inspire/base/static/js/search/search_results.js]: this will be removed.

File: inspire/base/static/js/search/search_results.js [hunk 2]
```diff
@@ -44,8 +44,215 @@ require(['jquery', 'bootstrap'], function($) {
...
+    var re = new RegExp('(?:\\?|&)' + key + '=(.*?)(?=&|$)', 'gi');
+    var r = [],
+      m;
+    while ((m = re.exec(document.location.search)) != null) r.push(m[1]);
+    return r;
+  }
+
+  $('input[type=checkbox]').change(function() {
```
Comment (jmartinm) [inline inspire/base/static/js/search/search_results.js]: Better to put an id to select from

File: inspire/base/static/js/search/search_results.js [hunk 3]
```diff
@@ -44,8 +44,215 @@ require(['jquery', 'bootstrap'], function($) {
...
+    var $this = $(this);
+    if (this.checked) {
+      if ($this.attr("id") != 'checkbox-select-all') {
```
Comment (jmartinm) [inline inspire/base/static/js/search/search_results.js]: This will not be needed if the selector only matches the checkboxes close to each result.

**Comment by ioannistsanaktsidis** [comment]
The reason that we have the problem is that search results(search/results_base.html) are wrapped to this form:
```
 <form class="form-inline" name="" action="{{ url_for('search.dispatch') }}" method="post">
```
so every time that I fire my ajax call: 
```
javascript
$("#download-format").on('click', function() {
      console.log(getType(url));
     $.ajax({
        type: "POST",
        url: getType(url),
        data: {ids: uniqueIds},
        success: function(data) {
          console.log(data);
          var response_data = "text/plain;charset=utf-8," + encodeURIComponent(data);
          $("body").append('<a id="data-download" href="data:' + response_data + '" download="references.bib">download</a>');
          var trigger_element = document.getElementById('data-download');
          trigger_element.click();
          $("#data-download").remove();
        }
      });
     });
```
in the success function the data from the form are also added and it is becoming a mess.
Any ideas how to avoid this problem? Is there any specific reason that results are wrapped to a form?

**Pull request closed without merging**